cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceQueryAdmin.java

```diff
@@ -56,7 +58,8 @@ public DefaultNamespaceQueryAdmin(NamespaceStore nsStore) {
    * @throws NamespaceNotFoundException if the requested namespace is not found
    */
   @Override
-  public NamespaceMeta get(Id.Namespace namespaceId) throws Exception {
+  public NamespaceMeta get(Id.Namespace namespaceId) throws NamespaceNotFoundException, IOException,
+    UnauthenticatedException {
```

Comment:
We should have a handler in app fabric (or some system container) that specifically serves this GET request